src/main/java/it/pagopa/ecommerce/payment/methods/application/v1/PaymentMethodService.java

```diff
@@ -539,7 +539,10 @@ public Mono<NpgSessionDocument> updateSession(
                 .flatMap(document -> document.map(Mono::just).orElse(Mono.empty()))
                 .switchIfEmpty(Mono.error(new OrderIdNotFoundException(orderId)))
                 .flatMap(document -> {
-                    if (document.transactionId() != null) {
+                    // Session associated to the order is associated to a different transaction id,
+                    // not permitted
+                    if (document.transactionId() != null
+                            && !document.transactionId().equals(updateData.getTransactionId())) {
```

Comment:
this check can be simplified with 
!Object.equals(document.transactionId(), updateData.getTransactionId())
since request transactionId is a mandatory field so always != null.

I would add an error log too here logging checked variables values

Reply:
Error log added in 2ee2cae